packages/xo-web/src/xo-app/pool/tab-advanced.js

```
@@ -245,11 +245,20 @@ class ToggleHa extends Component {
 
   _onChange = async value => {
     if (value) {
-      const haSrs = await confirm({
-        body: <EnableHaModal srs={this.props.pool.haSrs ?? []} pool={this.props.pool} />,
-        title: _('poolEnableHa'),
-        icon: 'pool',
-      })
+      let haSrs
+      try {
+        haSrs = await confirm({
+          body: <EnableHaModal srs={this.props.pool.haSrs ?? []} pool={this.props.pool} />,
+          title: _('poolEnableHa'),
+          icon: 'pool',
+        })
+      } catch (err) {
+        // don't throw for confirm cancel
+        if (typeof err === 'undefined') {
```

What about ignoring `undefined` errors directly in [`Toggle`](https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-web/src/common/form/toggle.js#L31) the same way we do it with [`ActionButton`](https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-web/src/common/action-button.js#L125-L126)?

---

I thought about this, but the toggle is used in so many place that I fear the unexpected consequences if we do so

---

If we only swallow `undefined` errors and properly rethrow any other error, it should be safe, as the only case where we throw (reject) `undefined` is when a `confirm` modal is canceled. And it might also fix other instances of `confirm`s being used in a `Toggle`'s `onChange`

---

Ok, if you think it's safe then I'll do that